esphome/components/usb_device/usb_device.cpp

```diff
@@ -9,7 +9,21 @@ namespace usb_device {
 
 static const char *const TAG = "usb_device";
 
-void UsbDevice::setup() { USB.begin(); }
+void UsbDevice::setup() {
+#ifdef USE_VENDOR_ID
+  USB.VID(this->vendor_id_);
+#endif
+#ifdef USE_PRODUCT_ID
+  USB.PID(this->product_id_);
+#endif
+#ifdef USE_MANUFACTURER_NAME
```

Comment:
Thanks for PR. I would just check is string is empty instead of `#ifdef`.

Reply:
done